crates/context/config/src/client/env/proxy/query/proposals.rs

```diff
@@ -31,10 +34,34 @@ impl Method<Starknet> for ProposalsRequest {
     type Returns = Vec<Proposal>;
 
     fn encode(self) -> eyre::Result<Vec<u8>> {
-        todo!()
+        let mut bytes = Vec::with_capacity(64); // 2 * 32 bytes for two u32 parameters
+
+        // First parameter (offset): pad to 32 bytes
+        bytes.extend_from_slice(&[0; 28]); // 28 bytes of zeros
+        bytes.extend_from_slice(&(self.offset as u32).to_be_bytes()); // 4 bytes of actual value
+
+        // Second parameter (length): pad to 32 bytes
+        bytes.extend_from_slice(&[0; 28]); // 28 bytes of zeros
+        bytes.extend_from_slice(&(self.length as u32).to_be_bytes()); // 4 bytes of actual value
```

Comment:
surely we can leave this to a derive impl of `Encode` on `StarknetProposal`, right?